benchmark/lib/store.js

```diff
@@ -1,39 +1,142 @@
+import { DynamoDBDocument, TransactWriteCommand } from '@aws-sdk/lib-dynamodb';
+import { randomUUID } from 'crypto';
 
-export const storeScores = async (scores, key) => {
-  const region = process.env.BENCH_REGION || 'us-west-2';
-  const s3 = new S3Client({ region });
-  const bucket = process.env.BENCH_BUCKET || 'ffcloud';
-  key ||= process.env.BENCH_KEY || 'benchmarks/latest.jsonl';
+const allScores = [];
+let id = 0; // per parallel execution sequential id, combined with uuid
+
+let docClient;
+
+const getDocClient = () => {
+  if (!docClient) {
+    const client = new DynamoDBClient({});
+    docClient = DynamoDBDocument.from(client);
+  }
+  return docClient;
+}
+
+const registerCommit = async () => {
+  const commitsTable = process.env.BENCH_COMMITS_TABLE;
+  const lookup = process.env.BENCH_LOOKUP || 'default';
+
+  const now = new Date();
+  const timestamp = now.toISOString();
+  const date = timestamp.split('T')[0];
+
+  const commit = process.env.COMMIT || 'local';
+  const branch = process.env.BRANCH || 'unknown';
+
+  if (commit == 'local') {
+    logger.debug('Skipping registering commit in DynamoDB');
+    return;
+  }
+  const docClient = getDocClient();
 
-  let existing = [];
   try {
-    const params = {
-      Bucket: bucket,
-      Key: key,
-    };
-    const data = await s3.send(new GetObjectCommand(params));
-    const body = await streamToString(data.Body);
-    existing = body
-      .split('\n')
-      .filter((line) => !!line.trim())
-      .map((line) => JSON.parse(line));
-  } catch (e) {
-    if (e.name == 'NoSuchKey') {
-      // ignore
+    // Conditionally put commmit if not already there (transaction / lock)
+    const transactParams = {
+      TransactItems: [
+        {
+          Put: {
+            TableName: commitsTable,
+            Item: { lookup: "LOCK", timestamp: commit },
+            ConditionExpression: 'attribute_not_exists(lookup)',
+          }
+        },
+        {
+          Put: {
+            TableName: commitsTable,
+            Item: { lookup, commit, branch, timestamp, date },
+            ConditionExpression: 'attribute_not_exists(lookup)',
+          }
+        },
+      ]
+    }
+    await docClient.send(new TransactWriteCommand(transactParams));
+    logger.debug(`Registered commit ${commit}`)
+  } catch (error) {
+    if (error.name === 'TransactionCanceledException') {
+        logger.debug('Commit already registered, doing nothing.');
     } else {
-      throw e;
+      logger.warn(`Error registering commit: ${error}`);
     }
   }
+}
+
+const persistAllScores = async () => {
+  const scoresTable = process.env.BENCH_SCORES_TABLE;
+
+  const now = new Date();
+  const timestamp = now.toISOString();
+  const date = timestamp.split('T')[0];
+
+  const commit = process.env.COMMIT || 'local';
+
+  if (commit == 'local') {
+    logger.debug(`Skipping putting aggregate scores in DynamoDB, logging to debug ${allScores.length} rows instead`);
+    logger.debug(updated)
```

Comment:
IIRC, logger.debug expects a string, so you prob need to do `logger.debug(JSON.stringify(updated))`